src/raftstore/store/peer.rs

```diff
@@ -2319,11 +2327,21 @@ impl ReadExecutor {
         if self.snapshot.is_some() {
             return;
         }
-        let engine = self.engine.clone();
-        self.snapshot = Some(Snapshot::new(engine).into_sync());
-        // Reading current timespec after snapshot, in case we do not
-        // expire lease in time.
-        atomic::fence(atomic::Ordering::Release);
+
+        loop {
```

Comment:
do we really need `loop` here?

seems if we get option_snapshot after write and the option_snapshot is not `None`, we can use this option_snapshot directly.